modules/base/algorithm/mesh/meshcameraalgorithms.cpp

```diff
@@ -0,0 +1,80 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/algorithm/mesh/meshcameraalgorithms.h>
+#include <modules/base/algorithm/mesh/axisalignedboundingbox.h>
+#include <inviwo/core/network/networklock.h>
+#include <inviwo/core/datastructures/buffer/bufferramprecision.h>
+
+namespace inviwo {
+
+namespace meshutil {
+
+void centerViewOnMeshes(const std::vector<std::shared_ptr<const Mesh>>& meshes,
+                        CameraProperty& camera, float minMaxRatio) {
+    auto minmax = meshutil::axisAlignedBoundingBox(meshes);
+    auto newLookTo = 0.5f * (minmax.first + minmax.second);
+    // Locking to avoid multiple evaluations when changing properties
+    NetworkLock lock(&camera);
+    // Make sure the new value is not clamped
+    auto& lookTo = camera.lookTo_;
+    lookTo.set(newLookTo, glm::min(lookTo.getMinValue(), newLookTo / minMaxRatio),
+               glm::max(lookTo.getMaxValue(), minMaxRatio * newLookTo), lookTo.getIncrement());
+    // Adjust near/far planes if necessary
+    auto nearFar = computeNearFarPlanes(minmax, camera);
+    camera.setNearFarPlaneDist(nearFar.first, nearFar.second, minMaxRatio);
+}
+
+std::pair<float, float> computeNearFarPlanes(std::pair<vec3, vec3> worldSpaceBoundingBox,
+                                             const CameraProperty& camera, float farNearRatio) {
+    auto nearPlaneDist = std::numeric_limits<float>::max();
+
+    auto farPlaneDist = 0.f;
+
+    auto cameraDir = glm::normalize(camera.get().getDirection());
+
+    auto maxViewLength =
+        std::max(glm::distance(camera.lookFrom_.getMaxValue(), camera.lookTo_.get()),
+                 glm::distance(camera.lookFrom_.getMinValue(), camera.lookTo_.get()));
+    auto furthestViewPoint = camera.getLookFrom() - cameraDir * maxViewLength;
+    // Project min/max points onto view ray and compute distances to the most zoomed out view point
+    farPlaneDist =
+        std::max(glm::distance(furthestViewPoint,
+                               cameraDir * glm::dot(worldSpaceBoundingBox.first, cameraDir)),
+                 glm::distance(furthestViewPoint,
+                               cameraDir * glm::dot(worldSpaceBoundingBox.second, cameraDir)));
+    // Try to have a reasonable precision in the depth buffer
+    nearPlaneDist = std::max(glm::epsilon<float>(), farNearRatio * farPlaneDist);
```

Comment:
epsilon is a rather poor choice for a reasonable near clip value. This will most likely result in artifacts regarding the depth buffer. Suggesting a "larger" value, e.g. something larger than `1.0fe-4`.

Reply:
this is also different from what the function docs state...

Reply:
We would like to keep it as small as possible to not mess with data on smaller scales. The farNearRatio should take care that the precision implications are limited.
Changed it to use same as docs. Perhaps even let user specify but use a default value?

Reply:
The problem is with some datasets like the head, there the scene is suddenly larger than [1000, 1000, 1000]... Run quite often into problems with nearclip smaller than 0.1 while farclip is >100

Reply:
Could not detect any artifacts when using the MRI head data set